### PR TITLE
adds a sample app with full configuration showing how to use it

### PR DESCRIPTION
**Pull request by rcavalcanti**
I'm new to the npm toolchain and had to search for a while to understand how this library should work.

I build a small sample app to play around until I get it working. 

This PR includes that sample app as a sub-folder so others can use it as example.

Moreover, it also fixes the README file. The usage example use the `await` keyword, but it's only legal inside an `async function`.

### CODE REVIEW

File: sample/foo-typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "json-typescript",
```
Comment (bcherny) [inline sample/foo-typescript/package.json]: "json-schema-to-typescript-example"

**Comment by rcavalcanti** [comment]
Hi @bcherny, 

Thanks for reviewing. Just pushed the changes you requested.

**Comment by rcavalcanti** [comment]
@bcherny I did on more commit. I realized that I have added a 'Bar' and 'Baz' examples. It was  not my intention to push them.

I also changed the schema type from Foo to Person which is more in alignment with its fields

File: README.md [hunk 2]
```diff
@@ -55,9 +55,17 @@ export interface ExampleSchema {
 
 ## Usage
 
+Check [sample app](sample/foo-typescript) for more details.
```
Comment (bcherny) [inline README.md]: Update path

File: README.md [hunk 3]
```diff
@@ -55,9 +55,17 @@ export interface ExampleSchema {
...
+
 ```js
 import {compileFromFile} from 'json-schema-to-typescript'
 fs.writeFileSync('foo.d.ts', await compileFromFile('foo.json'))
+
+async function generate() {
+  fs.writeFileSync('dist/foo.d.ts', await compileFromFile('src/foo.json'))
+}
```
Comment (bcherny) [inline README.md]: Rm duplicate block

File: README.md [hunk 4]
```diff
@@ -55,9 +55,17 @@ export interface ExampleSchema {
...
+
+generate()
```
Comment (bcherny) [inline README.md]: rm

File: example/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  "name": "json-schema-to-typescript-example",
+  "version": "1.0.0",
+  "description": "",
+  "main": "dist/index.js",
+  "typings": "dist/index.d.ts",
+  "scripts": {   
+    "build": "npm run clean && npm run build-sources && npm run lint && node dist/index.js",
+    "clean": "rimraf dist",
+    "build-sources": "tsc -p ./tsconfig.json",
+    "lint": "tslint -c tslint.json src/*.ts",
```
Comment (bcherny) [inline example/package.json]: rm lint

File: example/package.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "prepublish": "npm run build"
+  },
+  
+  "dependencies": {
+    "@types/node": "4.0.30",
+    "tslint": "3.15.1",
```
Comment (bcherny) [inline example/package.json]: rm

File: example/tslint.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (bcherny) [inline example/tslint.json]: rm this file, no need for lint in example

File: README.md (hunk 3 above)
Comment (rcavalcanti) [inline README.md]: the duplicated block to remove it not the async function, but the one on line 62. 

It's not possible to call await outside a async function.

Comment (bcherny) [inline README.md]: The assumption is that the containing function is async. I made it the top level fn here for brevity, as most users already know that async can't be top level.

Your complete example in /example should suffice I think.

**Comment by bcherny** [comment]
Thanks for the contribution @rcavalcanti!

**Pull request merged**